src/VisualStudio/Core/Def/Implementation/ProjectSystem/InvisibleEditor.cs

```diff
@@ -145,9 +145,11 @@ public void Dispose()
             }
         }
 
+#pragma warning disable CA1821 // Remove empty Finalizers
```

Comment:
I added a comment here: https://github.com/dotnet/roslyn-analyzers/issues/1241#issuecomment-310740456. I'll fix the Roslyn code to have the correct #ifs but this also needs a fix in the analyzer to avoid the suppression.